File: MyTaskApplication/app/src/main/java/th/ac/kku/cis/lab/mytaskapplication/TaskActivity.java
package th.ac.kku.cis.lab.mytaskapplication;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.util.Log;
import android.view.MenuItem;
import android.view.View;
import android.widget.CompoundButton;
import android.widget.EditText;
import android.widget.Switch;

import com.google.firebase.database.DataSnapshot;
import com.google.firebase.database.DatabaseError;
import com.google.firebase.database.DatabaseReference;
import com.google.firebase.database.FirebaseDatabase;
import com.google.firebase.database.ValueEventListener;

import java.util.HashMap;
import java.util.Map;

public class TaskActivity extends AppCompatActivity {

    private DatabaseReference mDatabase;
    private EditText editTextTaskname;
    private Switch switchStatus;
    private boolean status;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_task);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);


        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
        getSupportActionBar().setDisplayShowHomeEnabled(true);

        mDatabase = FirebaseDatabase.getInstance().getReference();

        FloatingActionButton fab = findViewById(R.id.fabSave);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                SaveTask();
            }
        });

        editTextTaskname = findViewById(R.id.etTask);
        switchStatus = findViewById(R.id.switchStatus);
        status = false;
        switchStatus.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
                status = isChecked;
            }
        });

        Intent myIntent = getIntent(); // gets the previously created intent
        Bundle extras = myIntent.getExtras();
        if (extras != null) {
            String name = extras.getSerializable("name").toString();
            boolean status = (boolean) extras.getSerializable("status");

            editTextTaskname.setText(name);
            switchStatus.setChecked(status);

            getSupportActionBar().setTitle("Edit " + name + " Task");
        }else{
            getSupportActionBar().setTitle("Add new Task");
        }
    }
    private void SaveTask(){
        String s = this.editTextTaskname.getText().toString();
        myTask newTask = new myTask(s,status);

        mDatabase.child(newTask.task).setValue(newTask);
        mDatabase.addValueEventListener(new ValueEventListener() {
            @Override
            public void onDataChange(@NonNull DataSnapshot dataSnapshot) {
                Log.i("Add New Data","Done");
                finish();
            }

            @Override
            public void onCancelled(@NonNull DatabaseError databaseError) {
                Log.i("Add New Data","Cancelled");
            }
        });
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
        }
        return super.onOptionsItemSelected(item);
    }

}

File: Lab03-Listview/app/src/main/java/th/ac/kku/cis/lab/listview/TaskActivity.kt
package th.ac.kku.cis.lab.listview

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.ListView
import android.widget.Toast
import th.ac.kku.cis.lab.listview.Adapter.TaskAdapter
import th.ac.kku.cis.lab.listview.Model.Task

class TaskActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_task)

        val tasks_list = ArrayList<Task>()
        val taskAdapter = TaskAdapter(this,R.layout.task_list,tasks_list)
        val listView : ListView = findViewById(R.id.listView_task)
        listView.adapter = taskAdapter

        val addButton : Button = findViewById(R.id.addtaslbutton)
        addButton.setOnClickListener{ view ->
            val newTesk : EditText = findViewById(R.id.newTask)
            val newCategory : EditText = findViewById(R.id.newCategory)

            tasks_list.add(Task(newTesk.text.toString(), newCategory.text.toString(), R.mipmap.ic_launcher ))
            taskAdapter.notifyDataSetChanged()
        }

        listView.setOnItemClickListener{parent, view, position, id ->
            val selectedItem = parent.getItemAtPosition(position) as Task
            Toast.makeText(this,selectedItem.name,Toast.LENGTH_SHORT).show()
        }

        listView.setOnItemLongClickListener { parent, view, position, id ->
            tasks_list.removeAt(position)
            taskAdapter.notifyDataSetChanged()
            true
        }
    }
}
